### PR TITLE
Clean some deprecated code guards

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
There is more of these guards to clean, this PR is jut a beginning.

### CODE REVIEW

File: apps/apps.c [hunk 1]
```diff
@@ -37,9 +37,7 @@
 #ifndef OPENSSL_NO_ENGINE
```
Comment (richsalz) [inline apps/apps.c]: this wrapper can also go.  it is always safe to include a header (see CHANGES)

**Comment by richsalz** [comment]
We still support building without MD5.  Not sure if that's something we want to keep.

**Comment by FdaSilvaYY** [comment]
Following 7ec8de16fe23e207d3e1bbb95d53843dbc9c06c8, it's seem no more the case :
`[me@localhost openssl]# ./config no-md5`
`Operating system: x86_64-whatever-linux2`
`Configuring OpenSSL version 1.1.1-dev (0x0x10101000L)`
`***** Unsupported options: no-md5``

```

File: apps/apps.c (hunk 1 above)
Comment (FdaSilvaYY) [inline apps/apps.c]: OK

**Comment by richsalz** [comment]
Okay, I was wrong.  Our internal test script seems to try it tho.

**Comment by mattcaswell** [comment]
We should update INSTALL...no-md5 is still listed as an option. Also I'm in two minds about whether we should remove these guards or not. They were broken for 1.1.0, and fixing them was too hard and too disruptive in the 1.1.0 timescales. However I could see that in the future we might want to be able to disable these things (no-md5 in particular).

**Comment by FdaSilvaYY** [comment]
@mattcaswell : Ok, I going to reduce these changes to the minimum : NO_RSA guards only .

**Comment by mattcaswell** [comment]
Well even NO_RSA I could foresee a time when we want to enable that.

**Pull request closed without merging**